### DISCUSSION TITLE
"ArraySetNumber3" not updating

### DISCUSSION THREAD

**Discussion by brainw0rm** [cables-gl/cables_docs#241] [! Welcome - readme first!]
with an animated array as input ->
"ArraySetNumber" node is correctly updating the output array with logged changes at index and at main loops fps while:
"ArraySetNumber3" does not update with the main loop. It just outputs an static array when triggered.

**Comment by steam0r** [comment]
hey, not really sure i get what you are trying to do, can you please provide a simple example describing the problem?

this seems to do what i think it should be doing:

![Bildschirmfoto 2022-05-18 um 10 59 58](https://user-images.githubusercontent.com/151271/169000884-9de783ef-c890-4e84-b97f-77d2a86284d3.png)

**Reply by brainw0rm** [reply]
Hey there!

If I feed the same animated array into both of them, "arraysetnumber" outputs an array which is updated 60 times per second while
the "arraysetnumber3" outputs a static array which is not updated with 60fps. You can see it if you hit the live inspector on the output. I cant show it in a picture. The array in the inspector in the following picture is not changing at all.

![image](https://user-images.githubusercontent.com/105754406/169848721-57ea24be-a279-41b5-bea7-1675a8c60d31.png)

**Reply by steam0r** [reply]
hm, okay, a patch with a simple example including your animation of the arrays would still be useful, as you can see from my setup the array is/should be changing or at least is updated. so a bit more context would help. from looking at your screenshot i am seeing that you are not changing the number on arraysetnumber3 at all, while you seem to do for the other ones.

**Reply by steam0r** [reply]
the code for these two ops are almost identical, as you can see when pressing E, are you sure your input array is changing?

**Reply by brainw0rm** [reply]
hey! sorry for my bad form :)
here we go: https://cables.gl/p/628bbcaac1bf42b3f098601d?s=SyB3rjRCROVb4KJc

![image](https://user-images.githubusercontent.com/105754406/169870649-c48ba702-6931-4d14-b8d2-b2fcb9e27a6c.png)

**Reply by steam0r** [reply]
all good, thanks! you need to make that public or invite me ("stephan") to it, though ;)

**Reply by steam0r** [reply]
have you tried using a sequence op to make sure the array set number is called AFTER you to the transformarray?

**Reply by brainw0rm** [reply]
I sent you an invite :) !
Yes, the sequence op doesnt make any difference.

**Reply by steam0r** [reply]
connecting the trigger to the reset-port instead of exec will update your array. if it's trigger via exec or by changing the array it's only comparing the lenght of the array (to save time) which is not changing in your example

**Reply by steam0r** [reply]
aaand, i guess your timer is not updating at 60fps either, so you might use that reset port to trigger the updating of the array a lot less, as it might affect performance....this of course depends on your final setup and not this example case

**Reply by steam0r** [reply]
maybe like this

![Bildschirmfoto 2022-05-23 um 19 22 34](https://user-images.githubusercontent.com/151271/169874014-f415af8c-4b02-468e-9560-347d4b5802f0.png)

**Reply by brainw0rm** [reply]
ahhh ok thank you for pointing that out :). sorry for creating a hustle, I didnt even consider using the reset trigger since on the arraysetnumber its also the execute one that does the job. Should have tried that first before posting :D. I come from using vvvv where the reset pin mostly functions in a different way. Thanks for helping out! Should I close this topic to avoid further confusion for other peeps? Cheers again!
